app/services/payout/service.rb

```diff
@@ -15,7 +15,14 @@ def create_message(payout_report:, publisher:, message:)
     # Internal: Creates entries for any reason for not paying out a publisher.
     #
     # Returns a boolean
-    def skip_publisher?(payout_report:, publisher:)
+    def skip_publisher?(payout_report:, publisher:, allowed_regions: [], connection: nil)
+      if !allowed_regions.blank? && connection.present?
+        if !allowed_regions.include?(connection.country&.upcase)
```

Comment:
There should be a method `valid_country?` on all the connection models.  You could use that if you don't want to have to pass the allowed regions through the various services.  I think it's also fine as is, though.

Reply:
For this service, we need to run the allowed countries call once so that we don't hit the API or even cache more than once. So we do that at the top of the service and pass the results down through the services. This makes for easier dependency injection testing and also allows us not to n+1 the calls. That method does seem convenient, though!